assets/js/character.js

```diff
@@ -1,69 +1,100 @@
-function Person(first, last) {
-  this.firstName = first;
-  this.lastName = last;
-}
+// end turn after  attack
 
-var h = new Person('Homer', 'Simpson');
+function Character(options) {
+  options = options || {};
+  var hitPoints = options.hitPoints || 100;
+  this.weapons = options.weapons || {};
+  this.logDamage = function(damage) {
+    '.narration-window'.html(this + 'takes' + damage + 'points of damage');
+  };
 
-var withNewName = function(person) {
-  var cp = new Person(person.firstName, person.lastName);
+  this.takeDamage = function(damage) { hitPoints -=  damage; };
 
-  cp.firstName = 'y';
-  return cp;
-};
+  this.getAttackStrength = function(weaponName) {
+    if (this.weapons[weaponName]) {
+      return this.weapons[weaponName];
+    }
 
-h = withNewName(h);
+    return 5;
+  };
 
-// Do some code here...
+  this.on('attacked', function(amount) {
+    this.takeDamage(amount);
+    this.logDamage(amount);
+  });
 
-h.firstName = 'something else';
+  this.getHealth = function() {return hitPoints;};
+}
 
-h = withNewName(h);
+function Hero() {
+}
 
-function Character(options) {
-    options = options || {};
-    var hitPoints = options.hitPoints || 100;
-    this.weapons = options.weapons || {};
+Hero.prototype = _.extend({
+  constructor: Character,
 
-    this.takeDamage = function(damage) { hitPoints -=  damage; };
+  attack: function(hostile, weapon) {
+    hostile.trigger('attacked', this.getAttackStrength(weapon));
+  }
+}, Backbone.Events);
 
-    this.getAttackStrength = function(weaponName) {
-      if (this.weapons[weaponName]) {
-        return this.weapons[weaponName];
-      }
+Mal = new Hero({
```

Comment:
When creating a variable, don't forget the `var` keyword. Forgetting `var` can make variables act a bit funny.